integration/test.ats

```diff
@@ -113,43 +124,86 @@ def tolerance_test(name, input_file, num_mpi_tasks=1, tolerance=0.00001):
+    restart_output_dir    = os.path.abspath("{0}_output_restart_{1}".format(name, restart_cycle))
+    restart_test_file     = os.path.join(restart_output_dir, "summary.json")
+    restart_baseline_file = os.path.abspath("{0}_restart_{1}.json".format(name, restart_cycle))
+
+    tolerance_arg = "--tolerance={0}".format(tolerance)
+    if ".json" in str(tolerance):
+        tolerance_arg = "--tolerance-file={0}".format(tolerance)
+
+    if should_baseline:
+        # Copy pre-existing normal serac summary file to baseline location
+        copy_args = "{0} {1}".format(normal_test_file, normal_baseline_file)
+        test(
+            executable=serac_copy,
+            clas=copy_args,
+            label=name,
+            nosrun=True)
 
-    if not should_rebaseline:
-        # Add driver test
-        test_args = "-i {0} -o {1}".format(input_file, output_dir)
-        my_test = test(
+        # Copy pre-existing restart serac summary file to baseline location
+        if restart_cycle is not None:
+            copy_args = "{0} {1}".format(restart_test_file, restart_baseline_file)
+            test(
+                executable=serac_copy,
+                clas=copy_args,
+                label="{0}_restart_{1}".format(name, restart_cycle),
+                nosrun=True)
+    else:
+        # Run serac normally
+        normal_test_args = "-i {0} -o {1}".format(input_file, normal_output_dir)
+        my_normal_test = test(
             executable=serac_path,
-            clas=test_args,
+            clas=normal_test_args,
             label=name,
             np=num_mpi_tasks)
 
-        # Add checker args
-        checker_args  = "--baseline={0} --test={1}".format(
-            baseline_summary_file, test_summary_file)
-
-        if ".json" in str(tolerance):
-            checker_args += " --tolerance-file={0}".format(tolerance)
-        else:
-            checker_args += " --tolerance={0}".format(tolerance)
-
-        # Add checker for driver test
-        my_checker = testif(
-            my_test,
+        # Tolerance test of normal baseline vs normal test
+        normal_checker_args = "--baseline={0} --test={1} {2}".format(
+            normal_baseline_file, normal_test_file, tolerance_arg)
+        my_normal_checker = testif(
+            my_normal_test,
             executable=serac_tolerance_checker,
-            clas=checker_args,
+            clas=normal_checker_args,
             label=name,
             nosrun=True)
-    else:
-        # Rebaseline
-        if os.path.exists(test_summary_file):
-            print("Rebaselining {0}".format(name))
-            shutil.copyfile(test_summary_file, baseline_summary_file)
-        else:
-            print("Warning: tried to copy test file that doesn't exist:\n{0}".format(
-                test_summary_file))
+
+        if restart_cycle is not None:
+            # Copy normal serac output to restart location
+            copy_args = "{0} {1}".format(normal_output_dir, restart_output_dir)
```

Comment:
This copying could become problematic down the line when our dumps get large.  We may want to add support for input directories.  Something like:

```
serac -i default.lua --input-dir=original_output_dir --output=new_output_dir
```

Reply:
https://github.com/LLNL/serac/issues/730#issue-1286561744